### ISSUE TITLE
Fedora: Unread notifications never going into notification dropdown

### ISSUE THREAD

**Issue by electricduck** [elementary/wingpanel-indicator-notifications#210]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
_'scuse me if this is supposed to be in [elementary/wingpanel-indicator-notifications](https://github.com/elementary/wingpanel-indicator-notifications/issues), or even somewhere else for non-elementaryOS issues (I couldn't quite work out who is in charge of packaging elementary stuff on Fedora)._

With Pantheon all set up on Fedora I noticed whenever I get a notification from an app, the notification bell doesn't do the usual thing of animating and showing a dot over it, nor does the notification itself go into the list after expiring (and the list itself always permanently states "No Notifications").

### To Reproduce
1. Install neccesary packages for a basic Pantheon/elementary install on Fedora
2. Send a notification from `notify-send`
3. Observe the notification not going into the notification list after expiring

### Expected behavior
Expired notification should go into notifications list to read later.

### Screenshots or screen recordings
![image](https://user-images.githubusercontent.com/1594644/119940365-fa40b680-bf86-11eb-8002-5b54c05dd0d3.png)
Notification from `notify-send` (I've also tried it with a few other apps)

![image](https://user-images.githubusercontent.com/1594644/119940420-0d538680-bf87-11eb-8607-5b82c0c839a8.png)
It never appears here!

## Platform Information
![image](https://user-images.githubusercontent.com/1594644/119940767-923ea000-bf87-11eb-9460-31e22eb618e2.png)
_(There's a color combo to make your eyes bleed)_

Versions I've tried
 - notifications
   - `0.11.20210311.giteec89e6.fc3` (package currently on Fedora, not that far behind the git)
 - wingpanel-indicator-notifications
   - `2.1.4-4.20210219git73f7da2.fc34` (package current on Fedora)
   - `47fec3e8dd28bafdde1040bf1959ecce69ea0fd6` (current git commit)

**Comment by decathorpe** [comment]
I am aware of this issue on Fedora but have not been able to debug it... It also happens with the latest nightly / git snapshot builds, so I don't think this is caused by the specific versions of things that are packaged on Fedora, but by some other difference between elementary OS and Fedora 😐

**Comment by tashimotor** [comment]
May be it caused by filters of dbus or access to dbus. I've debug a little, and found that the signal for adding notification in list never touched.

**Comment by decathorpe** [comment]
I've done some more digging. I found that the crash analytics service is getting up to 10 reports of wingpanel crashes per day: https://retrace.fedoraproject.org/faf/reports/104723/

Looking at the attached backtrace, the crash happens in `notifications_notification_monitor_get_instance` in the notifications indicator plugin, `/usr/lib64/wingpanel/libnotifications.so`.

The only thing that happens inside `NotificationMonitor.get_instance` is the call to the constructor, `new NotificationMonitor()`, where the backtrace doesn't show in which line the crash happens, though `g_log` does seem to indicate to me that it's caused by one of the `error()` calls in the `catch` blocks? Looks like something in the message filter setup or connection to dbus session bus fails, which would explain why just "nothing ever happens" with the indicator - it's not getting a DBus connection.

**Comment by davidmhewitt** [comment]
I did some digging into this and have opened #224 that I've tested to work with Fedora. The issue was with the indicator using an old deprecated way of watching the session bus for notifications, and Fedora presumably has a newer version of DBus or some stricter rules that prevent this deprecated way from working.

Either way, the PR switches over to the currently recommended way of watching the bus, so this should be some future proofing on elementary too.
